### ISSUE TITLE
[Venus] Train VGG16 on thyroid ultrasounds and apply transfer learning with cervix dataset

### ISSUE THREAD

**Issue by Domnyk** [Domnyk/pallas#19]
Assumptions 
- Untrained VGG16
- Thyroid ultrasound dataset consisting of 3644 samples. Thera are also extra 500 test samples but they do not include labels so can't be use in training. I'll probably need to perform augmentation since this dataset might be too small
  - What size of dataset is satisfactory?
- Train VGG16 on thyroid dataset 
  - TODO: how to define successful training? In terms of AUC? If yes, then what value of AUC is satisfactory?
- Perform transfer learning
  - Reset classification layer of VGG16 and use cervix dataset to train classification layer

Aspects to consider in future
- If performance of network would be poor, I can consider using pretrained VGG16

Expectations
- Can't formulate any expectations since I don't know what results to expect from thyroid dataset

**Comment by Domnyk** [comment]
** Initial results **
Network trained for 20 epochs on Thyroid dataset achieved AUC of 0.6927 on test set.

**Comment by Domnyk** [comment]
### Nearly final results

**Applied types of transfer learning**

- Classifier layers reset and conv layers freezed
- Classifier layers reset and last conv layer active
- Classifier layers reset and all conv layers active

**Two approaches**

- train/val/test split
- 10 fold cross validation

**Results**

- Networks trained on train/val/test for 20 epochs achieves good AUC on train set (0.82, 0.79, 0.88), worse on val set (0.63, 0.63, 0.72) and terrible on test set (0.43, 0.4, 0.48)
- Cross validation achieves better results than train/val/test approach. Best network achieves average of 0.6 AUC. However, for many folds train AUC is nearly random or worse than random.

**Next steps**

- Calculate AUC variance of folds in cross validation. Check if number of epochs is somehow related to variance of AUC
- Check if misclassified images of test set have something in common

**Comment by Domnyk** [comment]
### Final results

**AUC variance of folds in cross validation**
![stds](https://user-images.githubusercontent.com/1914955/88462827-83dab000-ceae-11ea-99a1-e7adf5e83dcb.png)
Conclusion: the bigger the number of epochs, the smaller the standard deviation of AUC

**Does misclassified images have something in common**
Probability achieved by all images is below 0.5. All images of preterm have probability below 0.5. This is only common thing between them. 

**I need to check what are the probability thresholds when AUC is calculate and see what output of model is then**

**Comment by Domnyk** [comment]
**Conclussions after consultations with thesis advisor**
1. Perhaps there is a bug in my code. I will try transfer learning on smaller datasets which are easier for neural nets to learn
2. Perhaps imbalance between preterm and control is a problem. I will try to use a subset of pregnancy dataset
